app/src/main/java/fi/helsinki/opintoni/integration/flamma/FlammaRestClient.java

```diff
@@ -19,23 +19,36 @@
 
 import com.google.common.collect.ImmutableMap;
 import com.rometools.rome.feed.atom.Feed;
+import fi.helsinki.opintoni.config.AppConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.feed.AtomFeedHttpMessageConverter;
+import org.springframework.stereotype.Component;
 import org.springframework.web.client.RestTemplate;
 import org.springframework.web.util.UriComponentsBuilder;
 
+import java.util.Collections;
 import java.util.Locale;
 
+@Component
 public class FlammaRestClient {
+    private final static Logger log = LoggerFactory.getLogger(FlammaRestClient.class);
 
     private final String baseUrl;
     private final RestTemplate restTemplate;
 
+    @Autowired
+    private AppConfiguration appConfiguration;
+
     private final ImmutableMap<String, String> studentFeedsByLocale;
     private final ImmutableMap<String, String> teacherFeedsByLocale;
-    private final ImmutableMap<String, String> studentOpenUniversityFeedsByLocale;
 
-    public FlammaRestClient(String baseUrl, RestTemplate restTemplate) {
-        this.baseUrl = baseUrl;
-        this.restTemplate = restTemplate;
+    @Autowired
+    public FlammaRestClient(AppConfiguration appConfiguration) {
+        this.baseUrl = appConfiguration.get("flamma.base.url");
+        this.restTemplate = initializeRestTemplate();
```

Comment:
Voisi ehkä tehdä sijoituksen `this.restTemplate`en tuolla `initializeRestTemplate`-metodissa, kun nythän toi metodi luo uuden restTemplaten, mutta ei alusta sitä ko. instanssille. Tai vaihtoehtoisesti nimeää ton initialize-metodin vaikka `createRestTemplate`ksi.